packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -0,0 +1,1136 @@
+       // a mix of both.
+       assert((boundaryMargin.horizontal.isInfinite
+           && boundaryMargin.vertical.isInfinite) || (boundaryMargin.top.isFinite
+           && boundaryMargin.right.isFinite && boundaryMargin.bottom.isFinite
+           && boundaryMargin.left.isFinite)),
+       super(key: key);
+
+  /// A margin for the visible boundaries of the child.
+  ///
+  /// Any transformation that results in the viewport being able to view outside
+  /// of the boundaries will be stopped at the boundary. The boundaries do not
+  /// rotate with the rest of the scene, so they are always aligned with the
+  /// viewport.
+  ///
+  /// To produce no boundaries at all, pass infinite [EdgeInsets], such as
+  /// `EdgeInsets.all(double.infinity)`.
+  ///
+  /// No edge can be NaN.
+  ///
+  /// Defaults to [EdgeInsets.zero], which results in boundaries that are the
+  /// exact same size and position as the [child].
+  final EdgeInsets boundaryMargin;
+
+  /// The Widget to perform the transformations on.
+  ///
+  /// Cannot be null.
+  final Widget child;
+
+  /// Whether the normal size constraints at this point in the widget tree are
+  /// applied to the child.
+  ///
+  /// If set to false, then the child will be given infinite constraints. This
+  /// is often useful when a child should be bigger than the InteractiveViewer.
+  ///
+  /// Defaults to true.
+  ///
+  /// {@tool dartpad --template=stateless_widget_scaffold}
+  /// This example shows how to create a table that is larger than the
+  /// InteractiveViewer. By setting constrained to false, the parts of the table
+  /// that are overflowing can be panned into view.
+  ///
+  /// ```dart
+  ///   Widget build(BuildContext context) {
+  ///     const int _rowCount = 20;
+  ///     const int _columnCount = 3;
+  ///
+  ///     return Scaffold(
+  ///       appBar: AppBar(
+  ///         title: const Text('Pannable Table'),
+  ///       ),
+  ///       body: InteractiveViewer(
+  ///         constrained: false,
+  ///         scaleEnabled: false,
+  ///         child: Table(
+  ///           columnWidths: <int, TableColumnWidth>{
+  ///             for (int column = 0; column < _columnCount; column += 1)
+  ///               column: const FixedColumnWidth(300.0),
+  ///           },
+  ///           children: <TableRow>[
+  ///             for (int row = 0; row < _rowCount; row += 1)
+  ///               TableRow(
+  ///                 children: <Widget>[
+  ///                   for (int column = 0; column < _columnCount; column += 1)
+  ///                     Container(
+  ///                       height: 100,
+  ///                       color: row % 2 + column % 2 == 1 ? Colors.red : Colors.green,
+  ///                     ),
+  ///                 ],
+  ///               ),
+  ///           ],
+  ///         ),
+  ///       ),
+  ///     );
+  ///   }
+  /// ```
+  /// {@end-tool}
+  final bool constrained;
+
+  /// If false, the user will be prevented from translating.
```

Comment:
nit: Is translating the same as panning? Maybe we should unify the terms in the docs and API?

Reply:
Good call.  I will use pan to refer to the user gesture and translate to refer to the matrix operation and make sure everything is consistent.